File: Backend/Phoenixia/src/main/java/com/cloud/phoenixia/model/FoodTruck.java
package com.cloud.phoenixia.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FoodTruck {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String menuName;

    private String imageUrl;

    private int price;

    private String description;

    @Enumerated(EnumType.STRING)
    private BoothStatus status;
}

File: Frontend/src/lib/types/foodtruck.ts
export interface MenuItem {
  name: string;
  description: string;
  price: number;
  imageUrl: string;
  status: string;
}

export interface FoodTruck {
  id: number;
  name: string;
  description: string;
  status: string;
  imageUrl: string;
  menuItems: MenuItem[];
}